### DISCUSSION TITLE
Plotting negative elevation angles for RHI scans

### DISCUSSION THREAD

**Discussion by scollis** [ARM-DOE/pyart#1270] [Q&A] [answered]
### Discussed in https://github.com/ARM-DOE/pyart/discussions/1253

<div type='discussions-op-text'>

<sup>Originally posted by **Ted-Whittock** August 31, 2022</sup>
Hello,

I have a bunch of RHI files from a mobile radar that was positioned along a mountain ridge, and so we have negative elevation angles to capture the precipitation over the surrounding lower terrain. (Iink to sample data: https://drive.google.com/drive/folders/19cQ3aS5qBt2gsQYjZu5cg14U_j7qt5J8?usp=sharing).

I have stitched 2 different azimuths together manually and added terrain heights below the radar (see attached image). However, I cannot figure out why it appears that the negative elevation angles (down to -1.4 deg) are not plotting. Is there a way to plot these negative elevations, or does Pyart always ignore them?

I am also aware that the y-axis on my attached plot is incorrect, but I wanted to address the issue of the negative elevation angles first.

Thanks,
Ted

![RHI_3panel_2angle_DBZHCandZDRCandCC_20170203_160235_to_20170203_160317_Azimuth65_and_245](https://user-images.githubusercontent.com/107871515/187554396-746a4759-7f27-408a-b3fd-c96d39d2151e.png)

</div>

**Comment by scollis** [comment]
@jfigui @wolfidan have you looked at negative RHI angles in PyRad?

**Comment by jfigui** [comment]
Hi @scollis,

We do have negative angles at MeteoSwiss but I never really paid attention. @wolfidan, can you have a quick look to that?

**Comment by wolfidan** [comment]
@scollis 
Sorry for the delay, I'm currently in vacation, I will take a look on thursday though,
Cheers

**Comment by jfigui** [comment]
Thanks a lot!

**Comment by mgrover1** [comment, marked as answer]
@Ted-Whittock I took a look at your sample file, and noticed that within the data quality parameters (antenna transition), they flag the negative elevation angles... by default, we filter out these transitions in the `plot_rhi` function, but you can change this behavior by setting

```python
filter_transitions=False
```

for example, I used the following snippet

```python
import pyart
import matplotlib.pyplot as plt

radar = pyart.io.read("cfrad.20170203_160235.683_to_20170203_160247.799_DOW7_v245_s03_az65.00_RHI.nc")

display = pyart.graph.RadarDisplay(radar)

display = pyart.graph.RadarDisplay(radar)
display.plot_rhi("DBZHCC", cmap='pyart_HomeyerRainbow', filter_transitions=False, edges=False)
plt.ylim(-5, 10)
```

which resulted in the following image
![Screen Shot 2022-09-14 at 4 53 44 PM](https://user-images.githubusercontent.com/26660300/190269890-d0fc2bed-9986-4e06-946e-0fadd87ff5bb.png)

**Reply by josha-rainmaker** [reply]
I was having the same issue. Tried the filter_transitions=False, but did not fix it. I had to set edges=False to get it to show the negative azimuths in my plots. Can anyone explain better what this "edges" parameter does? The Pyart documentation makes it seem like edges=True is the way it SHOULD be plotted (az, el, range is center of beam), regardless it should not be ignoring the negative elevations.

**Reply by zssherman** [reply]
Hi @josha-rainmaker , I'm going to dig into the code and investigate the reason why we ignore negative elevations when edges is True. These lines here are where its happening:
https://github.com/ARM-DOE/pyart/blob/26a73366bb2fdb1db2153ce345147f9672754d4a/pyart/core/transforms.py#L165

But the reason why im unsure of as this particular code predates my knowledge Py-ART, but my guess would be negative angles could possibly break the plotting or gridding code downstream.

**Reply by zssherman** [reply]
Forgot to answer the other portion of your question, but the edges parameter I believe is because Matplotlib pcolormesh expects grid boundary (edge) coordinates while radar data is usually stored as gate centers we have edges true to have the code interpolate/extrapolate edges from centers so every gate and the last ray are drawn fully for plotting. I will update the docs to clarify this

**Reply by josha-rainmaker** [reply]
@zssherman thanks for your help. That makes complete sense. That's why I was worried about setting edges=False because it seems like it would not correctly display the data based on my radar beam center angles I'm feeding it.

**Reply by zssherman** [reply]
One option might be to add a parameter to keep angles below the horizon so if keep_negative_angles=True
then ignore this line edges[edges < 0] = 0.0. I'll have to do more testing though.
